### ISSUE TITLE
0.76 -> 0.77 has issue with regex escape in YAML

### ISSUE THREAD

**Issue by PikachuEXE** [rubocop-hq/rubocop#7571]
I am using cop `Metrics/LineLength` with `Exclude`
```yaml
Metrics/LineLength:
  Enabled: true
  Severity: warning
  Max: 80
  IgnoredPatterns:
  # URLs
  - "http(s)?://"
  # Generated comment for db indexes
  - "#\\s+index_"
  # Long translation keys (breaking them up loses IDE auto linking)
  - "\\s+t\\(.[\\w\\.]+"
  - '\\s+t\\(\\n\\s+"[\\w\\.]+'
```
The last regex is to exclude
```ruby
# 2nd line is usually too long
t(
  "translation.key",
)
```
--------

## Expected behavior

No error raised

## Actual behavior

Error raised with message
`end pattern with unmatched parenthesis: /\\s+t\\(\\n\\s+"[\\w\\.]+/`

## Steps to reproduce the problem

- Include above config in rubocop.yml
- Run `bundle exec rubocop -Fd` (fail fast and debug)

## RuboCop version

```
$ bundle exec rubocop -V
0.77.0 (using Parser 2.6.5.0, running on ruby 2.6.5 x86_64-darwin18)
```

#### Stacktrace (truncated)
```
Scanning /Users/pikachuexe/projects/spacious/spacious-rails/app/resources/apps/spacious_website/pages/buildings/building_level_residential_for_sale_listing_search_page/meta_tags_data/cell.rb
An error occurred while Metrics/LineLength cop was inspecting /Users/pikachuexe/projects/spacious/spacious-rails/app/resources/apps/spacious_website/pages/buildings/building_level_residential_for_sale_listing_search_page/meta_tags_data/cell.rb.
end pattern with unmatched parenthesis: /\\s+t\\(\\n\\s+"[\\w\\.]+/
```

**Comment by buehmann** [comment]
In your config you have put this into single quotes:
```yaml
  - '\\s+t\\(\\n\\s+"[\\w\\.]+'
```
That means the backslashes are taken literally and you end up with the following regular expression, which clearly has an unmatched parenthesis:
```ruby
/\\s+t\\(\\n\\s+"[\\w\\.]+/
# end pattern with unmatched parenthesis: /\\s+t\\(\\n\\s+"[\\w\\.]+/
```
Therefore you need to fix your config. (By the way, if I remember correctly, `Metrics/LineLength` looks at one line at a time, so you will not have much luck with a multiline regexp.)

**Comment by PikachuEXE** [comment]
Thanks
Updated to use `"\\s+\"[a-z_\\.]+"`

I think multiline one works before 0.77 but I am too lazy to downgrade and test
